Add a "data" attribute to postcss_binary

nex3 opened the pull request.

nex3 committed:
Add a "data" attribute to postcss_binary

rzhw commented on internal/run.bzl:
```diff
@@ -90,6 +90,7 @@ postcss_run = rule(
         ),
         "output_name": attr.string(default = ""),
         "additional_outputs": attr.output_list(),
+        "data": attr.label_list(allow_files = True),
```
Is `data` not automatically consumed by Bazel for rules?

nex3 replied:
It's not for custom rules, only built-in ones.

rzhw commented on internal/binary.bzl:
```diff
@@ -31,6 +31,7 @@ def postcss_binary(
         additional_outputs = [],
         output_name = "",
         map_annotation = False,
+        data = [],
```
Is data not passed along by kwargs?

nex3 replied:
We need to make sure it only goes to `postcss_run`, not `postcss_gen_runner`.

rzhw reviewed (commented):
I'm surprised these changes are needed - is there something special about Bazel's handling of `data` that stops us from passing it along via **kwargs through to rules?

nex3 committed:
Add a test for the "data" attribute

nex3 commented:
I tried to add a test for this, but I can't get it working. The `header.txt` file is definitely being passed as an input to the `postcss_run` action, but it doesn't seem to show up in either its working directory or its runfiles. I'm going to need help figuring this one out.

nex3 committed:
Add special JS variables that can be accessed from plugin configs

This allows postcss_binary invocations to refer to data files and
additional outputs using the correct paths without needing to change
directories or manually write workspace paths. I considered using
Blaze's make variable substitution for this instead, but that doesn't
work for outputs: the substitution is done in postcss_gen_runner, at
which point Bazel doesn't know which files are outputs and so doesn't
know to return their output location.

rzhw commented on examples/additional_outputs/BUILD:
```diff
@@ -28,7 +28,7 @@ postcss_binary(
     src = "style.css",
     plugins = {
-        ":list_selectors": "[{markdownDir: 'examples/additional_outputs'}]",
+        ":list_selectors": "[{ output: bazel.additionalOutputs[0] }]",
```
This change would still keep this example viable, but to add some context, I modelled this after [postcss-svg-sprite](https://github.com/jostinsu/postcss-svg-sprite), which references directories instead of files.

A user could probably still achieve the same result with something like `require('path').dirname(bazel.additionalOutputs[0])`. That being said I'm happy to approve this PR and make cleaner dir support into its own issue.

nex3 replied:
They could also write `bazel.binDir + 'examples/additional_outputs'` which may be cleaner.
